server/metadata/namespace.go

```diff
@@ -0,0 +1,38 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+// NamespaceExtractor - extract the namespace from context
+type NamespaceExtractor interface {
```

Comment:
this interface if needed can also live in server package which has one implementation in the above class ^